src/containers/Experiment/DataSetStats.js

```diff
@@ -1,52 +1,120 @@
+import intersect from 'lodash/intersection';
+import union from 'lodash/union';
+import difference from 'lodash/difference';
+
 /**
- * Contains helper methods to calculate stats between a dataset and a list of samples
- * eg, how many processed samples are already in the dataset
+ * Contains several helper methods to perform operations on datasets. A dataset is a dictionary where
+ * the keys are experiment accession codes, and the values are a list of sample accession codes.
  */
-export default class DataSetStats {
-  constructor(dataSet, samples) {
-    this.dataSet = dataSet;
-    this.samples = samples;
+export class DataSetOperations {
+  /**
+   * Intersect two datasets
+   * @param {*} d1 dataset 1
+   * @param {*} d2 dataset 2
+   */
+  static intersect(d1, d2) {
+    const result = {};
+    const addedExperimentAccession = intersect(
+      Object.keys(d1),
+      Object.keys(d2)
+    );
+    for (let experimentAccession of addedExperimentAccession) {
+      result[experimentAccession] = intersect(
+        d1[experimentAccession],
+        d2[experimentAccession]
+      );
+    }
+    return result;
   }
 
-  anyProcessedSamples() {
-    return this.samples.some(x => x.is_processed);
+  /**
+   * Returns true if both given arguments have the same values, without taking into cosideration their order
+   * eg:
+   * sameItems([1,2], [2,1]) === true
+   * @param {Array} s1
+   * @param {Array} s2
+   */
+  static _sameItems(s1, s2) {
+    return difference(s1, s2).length === 0 && difference(s2, s1).length === 0;
+  }
+
+  static equal(d1, d2) {
```

Comment:
So this function tells whether two input objects have same keys and values? I wonder whether an implementation using `Map` in ES6 would be easier.

Reply:
What would you suggest? It should return true if both inputs have the same keys, and values without taking into account their order. Eg if:

```
d1 = { e1: ['s1', 's2']}
d2 = { e1: ['s2', 's1']}
```

`equal(d1, d2)` should be `true`

Reply:
I took a look at the methods in ES6 Map:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map
it is disappointing. I thought it offered some methods to compare two Map instances directly but I was wrong. I don't think `Map` or `Set` will make the implementation easier. Please feel free to leave the code as is.